cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -225,13 +225,14 @@ class device_row_comparator {
   {
     int last_null_depth = std::numeric_limits<int>::max();
     for (size_type i = 0; i < _lhs.num_columns(); ++i) {
-      int depth = _depth == nullptr ? std::numeric_limits<int>::max() : _depth[i];
+      int depth = _depth.has_value() ? _depth.value()[i] : std::numeric_limits<int>::max();
```

Comment:
The `optional::value` method performs error checking and will throw if `!has_value()`. This is working via relaxed constexpr which can have scary behavior: https://github.com/rapidsai/cudf/issues/7795

Just use `operator*` to retrieve the value w/o error checking`.